docs/components/Autocomplete/Web.stories.tsx

```diff
@@ -212,6 +237,422 @@ SectionHeading.args = {
+    contact: "(406) 555-0120 · seemore.b@gmail.com",
+    status: "Active",
+  },
+  {
+    value: 4,
+    label: "Sleeper Service",
+    address: "3891 Ranchview Dr. Richardson, California 62639",
+    contact: "(406) 555-0120",
+    status: "Active",
+  },
+];
+
+const CustomRenderingTemplate = () => {
+  const BasicCustomTemplate = () => {
+    const [value, setValue] = useState<Option | undefined>();
+    const [detailsValue, setDetailsValue] = useState<Option | undefined>();
+    const basicOptions = withDetailsOptions;
+    const sectionHeadingOptions = SectionHeadingOptions;
+    const planetSection = sectionHeadingOptions.find(
+      ({ label }) => label === "Planets",
+    )?.options;
+
+    return (
+      <Content>
+        <Autocomplete
+          placeholder="Search for a basic option"
+          value={value}
+          onChange={newValue => setValue(newValue)}
+          customRenderMenu={({
+            MenuWrapper,
+            inputFocused,
+            onOptionSelect,
+            options,
+            selectedOption,
+          }) => {
+            const { highlightedIndex } = useKeyboardNavigation({
+              options,
+              onOptionSelect,
+              visible: inputFocused,
+            });
+
+            return (
+              <MenuWrapper visible={inputFocused}>
+                {options.map((option, index) => {
+                  return (
+                    <MenuOption
+                      key={index}
+                      option={option}
+                      addSeparators={false}
+                      isHighlighted={index === highlightedIndex}
+                      onOptionSelect={onOptionSelect}
+                      isSelected={isOptionSelected(selectedOption, option)}
+                    />
+                  );
+                })}
+              </MenuWrapper>
+            );
+          }}
+          getOptions={getOptions}
+        />
+        <Autocomplete
+          placeholder="Search for an option with section headings"
+          initialOptions={sectionHeadingOptions}
+          value={detailsValue}
+          onChange={newValue => setDetailsValue(newValue)}
+          customRenderMenu={({
+            MenuWrapper,
+            inputFocused,
+            onOptionSelect,
+            options,
+            selectedOption,
+            menuRef,
+          }) => {
+            const { highlightedIndex } = useKeyboardNavigation({
+              visible: inputFocused,
+              options,
+              onOptionSelect,
+              menuRef,
+            });
+
+            return (
+              <MenuWrapper visible={inputFocused}>
+                {options.map((option, index) => {
+                  const optionStyle = getOptionStyling({
+                    option,
+                    highlightedIndex,
+                    options,
+                  });
+
+                  return (
+                    <MenuOption
+                      key={index}
```

Comment:
I ran into a non unique key problem in my own example and wanted to see how we avoided it in the webstory

I'm not sure this is a safe place to use the index since the options are not static. they could definitely change based on what we've filtered for with the search term.